### PR TITLE
Added implementation for Tizen.Net

### PR DESCRIPTION
**Pull request by maltegoetz**

### CODE REVIEW

**Comment by lothrop** [comment]
👍

**Comment by lothrop** [comment]
I've tested this with a Tizen.NET project. It requires Nuget 4.3 to install into a Tizen project. However, that is not yet available in VS 2017. Not sure how to test if this whole thing works and we've used the right identifier in the nuspec. nuget.exe 4.3.0-preview1 only extracts the package without modifying project.json or the csproj.

@onovotny, do you have any idea how we could test this?

**Comment by clairernovotny** [comment]
I'm not sure, maybe @rrelyea knows?

**Comment by jamesmontemagno** [comment]
Hey there! This is pretty cool and interesting. I am about to make huge breaking changes for V 3 and the APIs are gunna change, so I will see what I need to do to merge it in.

**Comment by jamesmontemagno** [comment]
What tooling do I need for VS 2017?

**Comment by maltegoetz** [comment]
VS 2017 is enough if you just want to compile it. If you want to try it, you need the VS tools for Tizen preview (https://developer.tizen.org/development/tizen-.net-preview/getting-started) they come with an emulator.

**Comment by jamesmontemagno** [comment]
Awesome thanks! will try it out, gunna take soem time with my new changes, will try to retarget to my other v3 branch

File: src/Plugin.Settings.Tizen/Settings.cs [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+                        Preference.Set(prefKey, Convert.ToString(value));
+                        break;
+                    case TypeCode.Int32:
+                        Preference.Set(prefKey, Convert.ToInt32(value, System.Globalization.CultureInfo.InvariantCulture));
+                        break;
+                    case TypeCode.Single:
+                        Preference.Set(prefKey, Convert.ToString(value, System.Globalization.CultureInfo.InvariantCulture));
+                        break;
+                    case TypeCode.DateTime:
+                        Preference.Set(Convert.ToString(-(Convert.ToDateTime(value)).ToUniversalTime().Ticks), prefKey);
```
Comment (JongHeonChoi) [inline src/Plugin.Settings.Tizen/Settings.cs]: Invalid parameter sequence.
 + Preference.Set(prefKey, Convert.ToString(-(Convert.ToDateTime(value)).ToUniversalTime().Ticks));

File: src/Plugin.Settings.Tizen/Settings.cs [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+                        break;
+                    default:
+                        if (value is Guid)
+                        {
+                            if (value == null)
+                                value = Guid.Empty;
+
+                            Preference.Set(((Guid)value).ToString(), prefKey);
```
Comment (JongHeonChoi) [inline src/Plugin.Settings.Tizen/Settings.cs]: Invalid parameter sequence.
 + Preference.Set(prefKey, ((Guid)value).ToString());

**Comment by JongHeonChoi** [comment]
src/Plugin.Settings.Tizen/Settings.cs
Invalid parameter sequence.
151 line => Preference.Set(prefKey, Convert.ToString(-(Convert.ToDateTime(value)).ToUniversalTime().Ticks));
159 line => Preference.Set(prefKey, ((Guid)value).ToString());

File: src/Plugin.Settings.Tizen/Settings.cs (hunk 2 above)
Comment (lothrop) [reply src/Plugin.Settings.Tizen/Settings.cs]: Thanks!

**Comment by clairernovotny** [comment]
@jamesmontemagno @maltegoetz if you convert this to csproj, you can now build the `tizen30` and `tizen40` TFM if you include my `MSBuild.Sdk.Extras` package. 

Instructions are here https://github.com/onovotny/MSBuildSdkExtras#targeting-tizen

This work much more cleanly than `project.json`. All uses of `project.json` should be removed since it's buggy.

**Comment by jamesmontemagno** [comment]
Is it possible yet for the nuget support for Tizen?

**Comment by clairernovotny** [comment]
yes, `tizen30` and `tizen40` are valid TFM's and my MSBuild.Sdk.Extras project lets you multi-target to them in an SDK style project.

**Comment by lothrop** [comment]
I’ll schedule working on this. Will AppVeyor be able to build it?

**Comment by clairernovotny** [comment]
Should be since Tizen is basically .NET Standard 1.6 plus their libraries. Last I checked, there's no local tooling to install.

**Comment by jamesmontemagno** [comment]
Yeah good question about appveyor... probably not i guess....

**Comment by clairernovotny** [comment]
@jamesmontemagno AppVeyor should work. no special tooling is required, it's just library/nuget references.

**Comment by lothrop** [comment]
@onovotny I'm not sure if I should be targeting tizen30 or tizen40. The Nuget package Tizen.NET seemed to be only available as 3.0.0.

**Comment by lothrop** [comment]
All set to merge.

**Comment by clairernovotny** [comment]
Why are you using a nuspec and not msbuld /t:pack with an SDK style project?

**Comment by clairernovotny** [comment]
I guess converting to SDK style projects could be a separate PR -- but these can all be one project with multi-targeting:
image

**Comment by jamesmontemagno** [comment]
@onovotny I don't want to use any of that new stuff, I haven't had time to actually learn all that. This works and I like to be explicit. 

I also don't want to use multi-targeting at all and want them separate :)

**Pull request merged**